src/engine/IdTable.h

```diff
@@ -624,9 +638,9 @@ class IdTableDynamicIterator {
 
 // Common interface specialication for the dynamic (number of columns only known
 // at runtime) IdTable
-template <typename DATA>
-class IdTableImpl<0, DATA> {
-  template <int INCOLS, typename INDATA>
+template <typename DATA, typename Alloc>
+class IdTableImpl<0, DATA, Alloc> {
+  template <int INCOLS, typename INDATA, typename INALLOC>
```

Comment:
We have two different template parameters, which both specify the number of Columns, the "main" parameter is always called `COLS`.
When we e.g. want to make the template instantiations of `IdTable` with different `COLS` friends with each other, in this friend definition the "other" ID tables column parameter must be called differently, so `COLS` is already taken. Currently this is inconsistent between `INCOLS` (e.g. when a IdTable with COLS is constructed from an IdTable with `INCOLS`) and 'NEW_COLS' for the other direction. So it is necessary to use a different name, but it could e.g. always be `OTHER_COLS` (the terminology of "other" vs. "this" is somewhat spread for this type of things). What do you think of this suggestion (Same goes for all Other Parameters as well)

Reply:
I think the OTHER_ prefix is a good idea